astropy/io/votable/tests/table_test.py

```diff
@@ -95,14 +95,19 @@ def test_read_through_table_interface(tmpdir):
 
 
 def test_read_through_table_interface2():
-    from astropy.table import Table
-
     with get_pkg_data_fileobj('data/regression.xml', encoding='binary') as fd:
         t = Table.read(fd, format='votable', table_id='last_table')
 
     assert len(t) == 0
 
 
+def test_pass_kwargs_through_table_interface():
+    # Table.read() should pass on keyword arguments meant for parse()
+    filename = get_pkg_data_filename('data/nonstandard_units.xml')
+    t = Table.read(filename, format='votable', unit_format='generic')
+    assert not isinstance(t['Flux1'].unit, UnrecognizedUnit)
```

Comment:
Isn't it more explicit to test that the unit is actually what you want it to be, rather than simply not `UnrecognizedUnit`?

Reply:
I have updated the test.